packages/react-native/src/executors/run-ios/run-ios.impl.ts

```diff
@@ -28,77 +20,69 @@ export default async function* runIosExecutor(
     context.projectsConfigurations.projects[context.projectName].root;
-  ensureNodeModulesSymlink(context.root, projectRoot);
-  if (options.sync) {
-    displayNewlyAddedDepsMessage(
-      context.projectName,
-      await syncDeps(
-        context.projectName,
-        projectRoot,
-        context.root,
-        context.projectGraph
-      )
-    );
-  }
 
-  if (options.install) {
-    await podInstall(
-      join(context.root, projectRoot, 'ios'),
-      options.buildFolder
+  if (options.packager && options.mode !== 'Release') {
+    const startResults = startExecutor(
+      {
+        port: options.port,
+        resetCache: options.resetCache,
+        interactive: true,
+      },
+      context
     );
-  }
-
-  try {
-    const tasks = [runCliRunIOS(context.root, projectRoot, options)];
-    if (options.packager && options.mode !== 'Release') {
-      tasks.push(
-        runCliStart(context.root, projectRoot, {
-          port: options.port,
-          resetCache: options.resetCache,
-          interactive: options.interactive,
-        })
-      );
-    }
-
-    await Promise.all(tasks);
-
-    yield { success: true };
-  } finally {
-    if (childProcess) {
-      childProcess.kill();
+    for await (const result of startResults) {
+      if (result.success) {
+        await runCliRunIOS(context.root, projectRoot, options);
+        yield {
+          success: true,
+        };
+      } else {
+        return result;
+      }
     }
+  } else {
+    await runCliRunIOS(context.root, projectRoot, options);
   }
+
+  yield { success: true };
 }
 
 function runCliRunIOS(
   workspaceRoot: string,
   projectRoot: string,
   options: ReactNativeRunIosOptions
-) {
-  return new Promise((resolve, reject) => {
+): Promise<ChildProcess> {
+  return new Promise<ChildProcess>((resolve, reject) => {
     /**
      * Call the react native cli with option `--no-packager`
      * Not passing '--packager' due to cli will launch start command from the project root
      */
-    childProcess = fork(
-      join(workspaceRoot, './node_modules/react-native/cli.js'),
+    const childProcess = fork(
+      require.resolve('react-native/cli.js'),
       ['run-ios', ...createRunIOSOptions(options), '--no-packager'],
       {
-        cwd: join(workspaceRoot, projectRoot),
+        stdio: 'inherit',
+        cwd: pathResolve(workspaceRoot, projectRoot),
         env: { ...process.env, RCT_METRO_PORT: options.port.toString() },
       }
     );
 
-    // Ensure the child process is killed when the parent exits
-    process.on('exit', () => childProcess.kill());
-    process.on('SIGTERM', () => childProcess.kill());
+    /**
+     * Ensure the child process is killed when the parent exits
+     */
+    const processExitListener = (signal?: number | NodeJS.Signals) => () =>
+      childProcess.kill(signal);
+    process.on('exit', processExitListener);
+    process.on('SIGTERM', processExitListener);
+    process.on('SIGINT', processExitListener);
```

Comment:
Same here, need to call `process.exit()` manually on `SIGINT` or else the process will not exit.

Reply:
added